### PR TITLE
conditionally enabling dataflow for ImportExampleGen

### LINKED ISSUE #20: ImportExampleGen integrated with Dataflow
The default VM running `ImportExampleGen` step of Vertex Pipeline does not have sufficient resources to handle the entire raw Sidewalk datasets. Hence, we need to integrate `Dataflow` into the `ImportExampleGen` component.

In order to do this, we can call [`with_beam_pipeline_args()`](https://www.tensorflow.org/tfx/api_docs/python/tfx/v1/components/ImportExampleGen) method on the `ImportExampleGen` with appropriate configurations of Dataflow:

```
beam_pipeline_args = [
    "--runner=DataflowRunner",
    "--region=" + GOOGLE_CLOUD_REGION,
    "--service_account_email=" + DATAFLOW_SERVICE_ACCOUNT,
    "--machine_type=" + MACHINE_TYPE,
    "--experiments=use_runner_v2",  
    "--max_num_workers=" + str(max_num_workers),
    "--disk_size_gb=" + str(disk_size),
]

example_gen = ImportExampleGen(...)
example_gen.with_beam_pipeline_args(beam_pipeline_args)
```

### PR DESCRIPTION
**Pull request by deep-diver**
Closes #20 

This PR includes the following changes:
- by scanning a environment variable `FULL_RES_DATA`
  - it will choose the GCS bucket where the full resolution of images are
  - it will choose different training function in `model_full_res.py`
  - it will choose dataflow specific configurations, then insert it to `ImportExampleGen` with `with_beam_pipeline_args` method

the `FULL_RES_DATA` environment variable is controlled by GitHub Action. You can provide the value `true` or `false` within GitHub Action panel. 

스크린샷 2022-09-19 오전 10 02 59

### CODE REVIEW

File: .github/workflows/cd-traing-pipeline.yml [hunk 1]
```diff
@@ -29,11 +33,16 @@ jobs:
...
+        python-version: '3.9'
         
     - name: 'Install required Python packages'
       run: |
         pip install -r requirements.txt
+
+    - name: 'Install cpp Protocol Buffer'
+      run: |
+        export PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=cpp
+        sudo pip install --upgrade --force-reinstall protobuf==3.17.3
```
Comment (sayakpaul) [inline .github/workflows/cd-traing-pipeline.yml:45]: Why this one needed?

File: training_pipeline/models/model_full_res.py [hunk 2]
```diff
@@ -0,0 +1,300 @@
+from typing import List, Dict, Tuple
+
+import absl
+import tensorflow as tf
+from tensorflow.keras.optimizers import Adam
+from tfx.components.trainer.fn_args_utils import FnArgs
+
+_CONCRETE_INPUT = "pixel_values"
+_RAW_IMG_SIZE = 256
```
Comment (sayakpaul) [inline training_pipeline/models/model_full_res.py:9]: I guess it's not needed?

File: training_pipeline/models/model_full_res.py [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+"""
+    _build_model builds a UNET model. The implementation codes are
+    borrowed from the [TF official tutorial on Semantic Segmentation]
+    (https://www.tensorflow.org/tutorials/images/segmentation)
+"""
+
+
+def _build_model(num_labels: int) -> tf.keras.Model:
+    base_model = tf.keras.applications.MobileNetV2(
+        input_shape=[128, 128, 3], include_top=False
```
Comment (sayakpaul) [inline training_pipeline/models/model_full_res.py:149]: 128 is a magic number here. Consider replacing it with `_INPUT_IMG_SIZE`.

**Review by sayakpaul — CHANGES_REQUESTED** [review]
Looking great. Minor changes.

File: .github/workflows/cd-traing-pipeline.yml (hunk 1 above)
Comment (deep-diver) [reply .github/workflows/cd-traing-pipeline.yml:45]: ah sorry about this one.
at the time, GitHub Action complained about something about protocol buffer, but seems like working OK now. will remove this one.

**Comment by deep-diver** [comment]
@sayakpaul 

addressed your comments! thanks for the careful inspection :)

**Pull request merged**